### ISSUE TITLE
Deploy of a new version fails with: "tar: invalid tar magic " after Deis update 1.5 to 1.7

### ISSUE THREAD

**Issue by dpanelli** [deis/deis#3871]
Today we updated our development platform form deis 1.5 to 1.7.2.
After the update the push of play  application fails  after image is built with:

Successfully built 8aa9c3adcbbe
-----> Pushing image to private registry

tar: invalid tar magic

Adding a Procfile does not mitigate the problem.

Play Version 2.4 , Jvm 1.8, deis deployed on Ec2.

**Comment by ineu** [comment]
See the same on 1.7.1. Restarted deis-registry and deis-builder, remove /home/git/*.git inside of deis-builder, no success

**Comment by Joshua-Anderson** [comment]
@ineu Did you try adding / do you have a `Procfile`?

Does the example app in your langauge deploy properly?

**Comment by ineu** [comment]
@Joshua-Anderson Sorry, I don't really understand the question.
I didn't do any changes to the app since upgrade to Deis 1.7. I'm upgrading to each version from 1.4, and this is the first time I see such problem with the same old application.

**Comment by Joshua-Anderson** [comment]
@ineu A [`Procfile` is a file that defines your web processes in deis](https://devcenter.heroku.com/articles/procfile). They're not necessary, as most buildpacks have default `Procfile` that they automatically use. However, other people have fixed this error by adding one themselves.

Deis has a bunch of example applications [here](https://github.com/deis). If they have one in the same language as the app that is failing to deploy, could you try it and see if it deploys.

**Comment by carmstrong** [comment]
> However, other people have fixed this error by adding one themselves.

I think the main concern from @ineu is that his app worked just fine on previous releases. 

@ineu Can you share anything about your application? Is it a play application like @dpanelli's?

**Comment by bacongobbler** [comment]
I'm personally interested to see if https://github.com/deis/deis/commit/835774225cd27d64425f23727075833d8cbc2b44 is what's causing the issue to occur. Adding debug output to builder by adding `set -x` to the top of `/etc/confd/templates/builder` inside the builder should help expose where this issue is coming from.

**Comment by mboersma** [comment]
I was able to reproduce this with the [example-play](https://github.com/deis/example-play) app after I removed the `Procfile` and commited it:
```console
Removing intermediate container 5833da6445dc
Successfully built 590d7ce4129b
-----> Pushing image to private registry

tar: invalid tar magic
To ssh://git@deis.local3.deisapp.com:2222/quoted-waterbed.git
 ! [remote rejected] master -> master (pre-receive hook declined)
```

Here's what is happening with `set -x` added to `/etc/confd/templates/builder`:
```console
-----> Pushing image to private registry
+ docker push 172.17.8.100:5000/easier-valkyrie:git-44d9d691
+ echo

+ '[' -f Procfile ']'
+ '[' -f /home/git/easier-valkyrie.git/build/tmp.iC2NHtsDaM/slug.tgz ']'
+ tar -tf /home/git/easier-valkyrie.git/build/tmp.iC2NHtsDaM/slug.tgz ./Procfile
++ extract-types
++ tar --to-stdout -xf /home/git/easier-valkyrie.git/build/tmp.iC2NHtsDaM/slug.tgz ./.release
tar: invalid tar magic
+ PROCFILE='{}'
To ssh://git@deis.local3.deisapp.com:2222/easier-valkyrie.git
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'ssh://git@deis.local3.deisapp.com:2222/easier-valkyrie.git'
```

**Comment by ineu** [comment]
> Can you share anything about your application? Is it a play application like @dpanelli's?

Unfortunately not. It is production application, and I cannot share its source code.

> I think the main concern from @ineu is that his app worked just fine on previous releases. 

Exactly. Procfile shouldn't be mandatory. For example, I have separate 'app' which only contains rails migrations for the database shared by few other applications. As it doesn't have anything to run, it doesn't need Procfile.

**Comment by dpanelli** [comment]
We tried another application in node.js and it works.

**Comment by mboersma** [comment]
Did the working app have a Procfile? I think I found and fixed this issue in #3881.

**Comment by ineu** [comment]
The trick with Procfile fixes it. For our failing app we've added Procfile with some rubbish like
```
asd: zxc
```
so it doesn't really declare anything meaningful.
Unfortunately empty Procfile won't work, raising exception on YAML parse.

**Comment by dpanelli** [comment]
@mboersma the fix will be released in 1.8? when it's scheduled?

**Comment by bacongobbler** [comment]
#3881 is targeted to v1.8 as per the tagged milestone, so yes.
